### PR TITLE
feat: support modifiers prop for components

### PR DESCRIPTION
**Pull request by f3ltron**
I have a concern about modifiers.

Every providers can have really differents modifiers from one to another. Would not be better to have like:
- Each provider receive one option(s) object where are define all available modifiers
- He is doing its own check / test
- So with that you can delete the default one. 

I know it will not be like on props for fit etc but one object. In my opinion It will be clearer and more modular.

What do you think ?

 @pi0 what do you think?

### CODE REVIEW

File: src/runtime/components/nuxt-img.vue [hunk 1]
```diff
@@ -60,7 +61,8 @@ export default {
         format: this.format,
         quality: this.quality,
         background: this.background,
-        fit: this.fit
+        fit: this.fit,
+        options: this.options
```
Comment (pi0) [inline src/runtime/components/nuxt-img.vue:65]: We do not support `modifiers.options` it should be `...this.modifiers` i guess

File: src/runtime/components/nuxt-picture.vue [hunk 2]
```diff
@@ -63,6 +63,7 @@ export default {
     quality: { type: [Number, String], required: false, default: undefined },
     background: { type: String, required: false, default: undefined },
     fit: { type: String, required: false, default: undefined },
+    options: { type: Object, required: false, default: () => {} },
```
Comment (pi0) [inline src/runtime/components/nuxt-picture.vue:66]: Default is not needed

File: src/runtime/components/nuxt-picture.vue [hunk 3]
```diff
@@ -120,6 +121,7 @@ export default {
         format: this.format,
         quality: this.quality,
         background: this.background,
+        options: this.options,
```
Comment (pi0) [inline src/runtime/components/nuxt-picture.vue:124]: Same for `...this.modifiers` (and probably should be first)

File: src/runtime/components/nuxt-img.vue (hunk 1 above)
Comment (f3ltron) [reply src/runtime/components/nuxt-img.vue:65]: ...this.options you mean ?

File: src/runtime/components/nuxt-picture.vue (hunk 3 above)
Comment (f3ltron) [reply src/runtime/components/nuxt-picture.vue:124]: you mean ...this.options i guess and yea agreed

**Review by pi0 — APPROVED** [review]

**Pull request merged**